src/Jackett.Test/Utils/CookieUtilTests.cs

```diff
@@ -0,0 +1,115 @@
+        }
+
+        [Test]
+        public void CookieHeaderToDictionaryDuplicateKeys()
+        {
+            // cookie with duplicate keys and whitespace separator instead of ;
+            // this cookie is not valid according to the standard, but it occurs in Jackett because we are concatenating
+            // cookies in many parts of the code (and we are not doing it well). this is safe because the whitespace
+            // can't be part of the key nor the value.
+            var cookieHeader = "__cfduid=d6237f041586694295; __cf_bm=TlOng/xyqckk-TMen38z+0RFYA7YA= __cf_bm=test";
+            var expectedCookieDictionary = new Dictionary<string, string>
+            {
+                {"__cfduid", "d6237f041586694295"},
+                {"__cf_bm", "test"} // we always assume the latest value is the most recent
+            };
+            CollectionAssert.AreEqual(expectedCookieDictionary, CookieUtil.CookieHeaderToDictionary(cookieHeader));
+        }
+
+        [Test]
+        public void CookieHeaderToDictionaryMalformed()
+        {
+            // malformed cookies
+            var cookieHeader = "__cfduidd6237f041586694295; __cf_;bm TlOng; good_cookie=value";
+            var expectedCookieDictionary = new Dictionary<string, string> {{"good_cookie", "value"},};
+            CollectionAssert.AreEqual(expectedCookieDictionary, CookieUtil.CookieHeaderToDictionary(cookieHeader));
+        }
+
+        [Test]
+        [SuppressMessage("ReSharper", "CollectionNeverUpdated.Local")]
+        public void CookieHeaderToDictionaryNull()
+        {
+            // null cookie header
+            var expectedCookieDictionary = new Dictionary<string, string>();
+            CollectionAssert.AreEqual(expectedCookieDictionary, CookieUtil.CookieHeaderToDictionary(null));
+        }
+
+        [Test]
+        public void CookieDictionaryToHeaderGood()
+        {
+            // valid cookies with non-alpha characters in the value
+            var cookieDictionary = new Dictionary<string, string>
+            {
+                {"__cfduid", "d6237f041586694295"}, {"__cf_bm", "TlOng/xyqckk-TMen38z+0RFYA7YA="}
+            };
+            var expectedCookieHeader = "__cfduid=d6237f041586694295; __cf_bm=TlOng/xyqckk-TMen38z+0RFYA7YA=";
+            CollectionAssert.AreEqual(expectedCookieHeader, CookieUtil.CookieDictionaryToHeader(cookieDictionary));
+        }
+
+        [Test]
+        public void CookieDictionaryToHeaderMalformed1()
+        {
+            // malformed key
+            var cookieDictionary = new Dictionary<string, string> {{"__cf_=bm", "34234234"}};
+            try
+            {
+                CookieUtil.CookieDictionaryToHeader(cookieDictionary);
+                Assert.Fail("Malformed cookie must fail");
+            }
+            catch (FormatException e)
+            {
+                Assert.AreEqual("The cookie '__cf_=bm=34234234' is malformed.", e.Message);
+            }
```

Comment:
```suggestion
            Assert.Throws<FormatException>(
                () => CookieUtil.CookieDictionaryToHeader(cookieDictionary), "The cookie '__cf_=bm=34234234' is malformed.");
```

Reply:
You code is not working, it's not cheeking the exception message. The message in your code is the error shown if the assert fails. I did other thing.

Reply:
My mistake, I just saw the intellisense called the parameter `message` and assumed it meant checking the message was equal.